src/silverlabnwb/nwb_file.py

```diff
@@ -664,9 +671,25 @@ def read_functional_data(self, folder_path):
         # this was the implication from the previous version of the code, as it
         # always used the last value of roi_dimensions - but that may be a bug?)
         ch_data_shape = np.concatenate((roi_dimensions,
-                                       [len(all_rois), cycles_per_trial]))[::-1]
+                                        [len(all_rois), cycles_per_trial]))[::-1]
         self._write_roi_data(all_rois, len(trials), cycles_per_trial, ch_data_shape, folder_path)
 
+    def add_custom_silverlab_data(self):
+        SilverLabExtensionClass = get_class('SilverLabExtension', 'silverlab_extended_schema')
+        silverlab_extension = SilverLabExtensionClass(name='optophysiology',
+                                                      cycle_time=self.custom_silverlab_dict['cycle_time'],
+                                                      cycles_per_trial=self.custom_silverlab_dict[
+                                                          'cycles_per_trial'],
+                                                      frame_size=self.custom_silverlab_dict['frame_size'],
+                                                      imaging_mode=self.custom_silverlab_dict['imaging_mode'],
+                                                      silverlab_api_version=self.SILVERLAB_NWB_VERSION,
+                                                      pockels=self.custom_silverlab_dict['zplane_pockels']
+                                                      )
```

Comment:
I think this is fine for now, and I like that it is explicit, but if we include more things, it may end up too long. In that case, passing the whole dictionary with `**custom_silverlab_dict` might be more readable (would also require renaming `zplane_pockels` to `pockels` or vice versa). I don't know if that's needed yet though.